src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -258,23 +264,53 @@ func (s *service) Bootstrapped(ctx thrift.Context) (*rpc.NodeBootstrappedResult_
 }
 
 func (s *service) Query(tctx thrift.Context, req *rpc.QueryRequest) (*rpc.QueryResult_, error) {
+	var sp opentracing.Span
+	result, sp, err := s.query(tctx, req)
+
+	if sp != nil {
+		if err != nil {
+			sp.LogFields(opentracinglog.Error(err))
+		}
+		sp.Finish()
+	}
+
+	return result, err
+}
+
+func (s *service) query(tctx thrift.Context, req *rpc.QueryRequest) (*rpc.QueryResult_, opentracing.Span, error) {
+	var (
+		ctx context.Context
+		sp  opentracing.Span
+		ok  bool
+	)
+
+	ctx = tchannelthrift.Context(tctx)
+
+	ctx, sp, ok = ctx.StartTraceSpan(querySpanID)
```

Comment:
hm, how about starting the span in the `Query()` method. just to minimise the amount of observability code we're putting side by side with execution logic

Reply:
hm why do we need to return ok from `StartTraceSpan` ?

Reply:
We technically don't but I guess it's a little cleaner than doing a `sp != nil`